### DISCUSSION TITLE
Select dictionary item in Query

### DISCUSSION THREAD

**Discussion by R4cOOn** [datalust/seq-tickets#1426] [General]
I'm using the Serilog middleware logger for ASP.NET.

The following query
`select HttpContext.Route from stream limit 20`
gives me 
>[{Key: 'action', Value: 'Version'}, {Key: 'controller', Value: 'Home'}]
[{Key: 'action', Value: 'Version'}, {Key: 'controller', Value: 'Home'}]
[{Key: 'controller', Value: 'Home'}, {Key: 'action', Value: 'Index'}]

What I'd like is to be able to group by and select the "controller" and "action" values.  
Please note that the order is random so I can't use `HttpContext.Route[0]`.

Ideally I'd be able to write a query similar to
`select HttpContext.Route['controller'], HttpContext.Route['action'] from stream limit 20`
and get 
> Home, Version
Home, Version
Home, Index

Is this sort of query supported?

**Comment by nblumhardt** [comment]
Hi @R4cOOn,

I don't think there's any way to express this currently; your best bet, if you can, would be to modify the serialization of `Route` (e.g. with an `ILogEventEnricher`) on the client-side, so that it's logged as a regular structure/object value.

Since we recently did implement lamba functions in the Seq query syntax, though, in future we might consider enabling something like `Single(Route, |r| r.Key = 'controller').Value` to improve this 🤔
